projects/js-packages/publicize-components/src/components/connection/index.js

```diff
@@ -65,27 +71,23 @@ class PublicizeConnection extends Component {
 		// Genericon names are dash separated
 		const serviceName = name.replace( '_', '-' );
 
-		let toggle = (
+		const toggle = (
 			<ConnectionToggle
 				id={ fieldId }
-				className="jetpack-publicize-connection-toggle"
+				className={ styles[ 'connection-toggle' ] }
 				checked={ enabled }
 				onChange={ this.onConnectionChange }
-				disabled={ disabled }
+				disabled={ disabled || this.connectionIsFailing() || this.connectionNeedsReauth() }
```

Comment:
I discovered that we were unnecessarily wrapping this in a `Disabled` component, so I incorporated the extra checks into this `disabled` prop